### DISCUSSION TITLE
Did widget support get pushed back?

### DISCUSSION THREAD

**Discussion by riverar** [microsoft/WindowsAppSDK#2944] [Q&A]
At the last WinUI Community Call, it was mentioned that Windows App SDK 1.2 was _targeting_ the ability to create Widgets for Windows 11. (I understand _targeting_ to mean _no guarantees_.) Given that Windows App SDK 1.2 Preview 1 has now been released, and preview releases are supposed to represent everything that will ship in the subsequent release, does this mean Widget support has been pushed to 2023/WinAppSDK 1.3?

![image](https://user-images.githubusercontent.com/475132/188915606-dde96801-ba42-4f0e-bf9f-900aeed9e5bb.png)

**Comment by DarranRowe** [comment]
They have prior form in things appearing later in the previews. If you look at the previous preview releases, you'll notice that some features were added in preview 2 and preview 3. As an example, IIRC, the ability to run Windows App SDK applications unpackaged didn't come until 1.0 preview 2 or 3.
This happens quite a lot in other products too. As an example, for quite a while with Visual Studio, the major C++ changes normally came with preview 2. Even now, the STL changelog on GitHub has 17.4 preview 2 adding two C++23 library features and preview 3 adding 13 C++23 library features. So no, I would say that the previews aren't meant to fully represent the subsequent release. They are more like giving you a chance to preview the current work.

**Reply by riverar** [reply]
That's not quite what we voted on as a community (https://github.com/microsoft/WindowsAppSDK/discussions/637). We voted on a plan where features would start their lives in the `experimental` (long-lead) channel and get promoted during development into the `preview` (next-release) and `stable` channels, with various quality control gates in place to protect the channels. (See "[Bars for moving across a release channel](https://github.com/microsoft/WindowsAppSDK/discussions/637#discussion-3290419)".) This release cadence/plan allows developers (us) to see exactly what's in the pipeline and plan accordingly.

The bar, per doc, for the `preview` channel is:
* Feature will almost certainly be included in the next GA release (it's always possible we have to pull something last minute, but that should be the rare exception)
* Near-stable API (few changes should be happening now)
* Functionality and stability is ~97% there, just ironing out final small bugs.
* Docs ready to be updated to indicate feature is now in next release. Docs are a bit more polished.

So given that we haven't seen Widgets in `experimental` or `preview` thus far, I can't imagine a future preview release coming out that doesn't violate the above rules. (Or is Microsoft really going to argue here that dropping a whole new feature, that skipped experimental, is a small change?)

It looks like Microsoft has abandoned some of these plans and is instead working on long-lead releases in a _private_ repository with a "looks good to us" gate for the `preview` channel.

**Reply by DarranRowe** [reply]
Well, they have already violated those above rules with 1.2.
[This entire namespace](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.appnotifications.builder?view=windows-app-sdk-1.2) wasn't in the experimental release.
[This entire namespace](https://docs.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.windows.security.accesscontrol?view=windows-app-sdk-1.2) wasn't in the experimental release.
What's more the MediaPlayerElement and MediaTransportControls classes were not in the experimental release.
What's more, as already mentioned, they have also previously violated those rules. Mica/Desktop Acrylic came in a preview 3. Unpackaged application support came in a preview 2 or 3 for x86 and x64 where the arm64 support came in the corresponding stable. Heck multiple windows on the same thread came in a servicing release. 1.1 didn't even have an experimental release.
So, I would honestly say that they didn't even follow those plans in the first place.
